vamp-server/convert.cpp

```diff
@@ -556,8 +558,15 @@ readInput(string format, RequestOrResponse::Direction direction, bool &eof)
     if (format == "json") {
         string err;
         auto result = readInputJson(direction, err, eof);
-        if (err != "") throw runtime_error(err);
-        else return result;
+        const bool isRecognisedError = !result.success && result.errorText != "";
+        
+        // if the RequestOrResponse (result) has been populated with success=false and an error message
+        // then the server returned a well formed error, it is safe to return it for conversion 
+        // -- but if err is populated, something else has gone wrong
+        if (isRecognisedError || err == "") 
```

Comment:
This condition doesn't quite match the accompanying comment. To me the last bit of the comment ("but if err is populated...") implies that we're going to bail out if err is populated, even if isRecognisedError is true. That's not what the condition does, and the ambiguity is distracting enough that after reading code + comment a couple of times I'm still not quite sure whether it's right or wrong.